packages/wrangler/src/__tests__/kv.test.ts

```diff
@@ -1,76 +1,210 @@
+          { binding = \\"UnitTestNamespace\\", id = \\"some-namespace-id\\" }"
+        `);
+      });
     });
 
-    it("can list namespaces", async () => {
-      const KVNamespaces: { title: string; id: string }[] = [
-        { title: "title-1", id: "id-1" },
-        { title: "title-2", id: "id-2" },
-      ];
-      setMock(
-        "/accounts/:accountId/storage/kv/namespaces\\?:qs",
-        (uri, init) => {
-          expect(uri[0]).toContain(
-            "/accounts/some-account-id/storage/kv/namespaces"
-          );
-          expect(uri[2]).toContain("per_page=100");
-          expect(uri[2]).toContain("order=title");
-          expect(uri[2]).toContain("direction=asc");
-          expect(uri[2]).toContain("page=1");
-          expect(init).toBe(undefined);
-          return KVNamespaces;
-        }
-      );
-      const { stdout } = await runWrangler("kv:namespace list");
-      const namespaces = JSON.parse(stdout) as { id: string; title: string }[];
-      expect(namespaces).toEqual(KVNamespaces);
+    describe("list", () => {
+      it("should list namespaces", async () => {
+        const KVNamespaces: { title: string; id: string }[] = [
+          { title: "title-1", id: "id-1" },
+          { title: "title-2", id: "id-2" },
+        ];
+        setMock(
+          "/accounts/:accountId/storage/kv/namespaces\\?:qs",
+          (uri, init) => {
+            expect(uri[0]).toContain(
+              "/accounts/some-account-id/storage/kv/namespaces"
+            );
+            expect(uri[2]).toContain("per_page=100");
+            expect(uri[2]).toContain("order=title");
+            expect(uri[2]).toContain("direction=asc");
+            expect(uri[2]).toContain("page=1");
+            expect(init).toBe(undefined);
+            return KVNamespaces;
+          }
+        );
+        const { stdout } = await runWrangler("kv:namespace list");
+        const namespaces = JSON.parse(stdout) as {
+          id: string;
+          title: string;
+        }[];
+        expect(namespaces).toEqual(KVNamespaces);
+      });
     });
 
-    it("can delete a namespace", async () => {
-      let accountId = "";
-      let namespaceId = "";
-      setMock(
-        "/accounts/:accountId/storage/kv/namespaces/:namespaceId",
-        (uri, init) => {
-          accountId = uri[1];
-          namespaceId = uri[2];
-          expect(uri[0]).toEqual(
-            "/accounts/some-account-id/storage/kv/namespaces/some-namespace-id"
-          );
-          expect(init.method).toBe("DELETE");
-        }
-      );
-      await runWrangler(`kv:namespace delete --namespace-id some-namespace-id`);
-      expect(accountId).toEqual("some-account-id");
-      expect(namespaceId).toEqual("some-namespace-id");
+    describe("delete", () => {
+      it("should delete a namespace", async () => {
+        let accountId = "";
+        let namespaceId = "";
+        setMock(
```

Comment:
I don't think setMock should every not return anything. Could you verify what the actual api returns on deleting a namespace? I think it's null.

Reply:
Sure, I'll take a look at the API.
FWIW the code that triggers this request does nothing with the return value:

```ts
await cfetch<{ id: string }>(
    `/accounts/${config.account_id}/storage/kv/namespaces/${id}`,
    { method: "DELETE" }
);
```

Reply:
From stepping through the code in real life ™️  I can see that it does indeed return a `null` value.